### PR TITLE
pkg/kvstore: fix literal copies lock value from

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4919)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

File: pkg/kvstore/etcd.go [hunk 1]
```diff
@@ -260,7 +259,7 @@ func newEtcdClient(config *client.Config, cfgPath string) (BackendOperations, er
 		firstSession: firstSession,
 		lockPaths:    map[string]*lock.Mutex{},
 		controllers:  controller.NewManager(),
-		RWMutex:      clientMutex,
+		RWMutex:      lock.RWMutex{},
```
Comment (rlenglet) [inline pkg/kvstore/etcd.go]: Wouldn't it work to just remove the field here? A mutex should be created anyway.

**Review by rlenglet — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

File: pkg/kvstore/etcd.go (hunk 1 above)
Comment (aanm) [reply pkg/kvstore/etcd.go]: Oh really? Let's try it out.

**Review by rlenglet — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Pull request merged**